### PR TITLE
Fixes for watch Treatments; deprecate Manifest BIND_LISTENER.

### PR DESCRIPTION
**Pull request by kskandispersonal**
As well as some minor, performance improvements.  To conserve watch battery, watchface refresh defaults to every minute.

### CODE REVIEW

File: app/src/main/java/com/eveningoutpost/dexdrip/Home.java [hunk 1]
```diff
@@ -2442,15 +2444,15 @@ public static String extraStatusLine() {
             if (prefs.getBoolean("status_line_carbs", false)) {
```
Comment (jamorham) [inline app/src/main/java/com/eveningoutpost/dexdrip/Home.java]: Changes below improve the look of the extra status line or is there another reason?

File: app/src/main/java/com/eveningoutpost/dexdrip/Home.java [hunk 2]
```diff
@@ -3016,6 +3018,7 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 break;
             case R.id.action_open_watch_settings:
                 startService(new Intent(this, WatchUpdaterService.class).setAction(WatchUpdaterService.ACTION_OPEN_SETTINGS));
```
Comment (jamorham) [inline app/src/main/java/com/eveningoutpost/dexdrip/Home.java]: This looks like a significant bug fix

File: app/src/main/java/com/eveningoutpost/dexdrip/Models/BloodTest.java [hunk 3]
```diff
@@ -124,7 +124,7 @@ private BloodTestMessage toMessageNative() {
     private static final long CLOSEST_READING_MS = 30000; // 30 seconds
 
     public static BloodTest create(long timestamp_ms, double mgdl, String source) {
```
Comment (jamorham) [inline app/src/main/java/com/eveningoutpost/dexdrip/Models/BloodTest.java]: I thought this was already fixed, is this github not showing the changes properly?

File: app/src/main/java/com/eveningoutpost/dexdrip/Home.java (hunk 1 above)
Comment (kskandispersonal) [inline app/src/main/java/com/eveningoutpost/dexdrip/Home.java]: Just UI improvements to make it consistent with other status line elements where there is no space after the label.

File: app/src/main/java/com/eveningoutpost/dexdrip/Models/Calibration.java [hunk 4]
```diff
@@ -526,8 +526,9 @@ public static Calibration create(double bg, long timeoffset, Context context, bo
                     BgSendQueue.handleNewBgReading(bgReading, "update", context);
                     // TODO probably should add a more fine grained prefs option in future
                     calculate_w_l_s(prefs.getBoolean("infrequent_calibration", false));
-                    adjustRecentBgReadings(adjustPast ? 30 : 2);
                     CalibrationSendQueue.addToQueue(calibration, context);
+                    BgReading.pushBgReadingSyncToWatch(bgReading, false);
```
Comment (jamorham) [inline app/src/main/java/com/eveningoutpost/dexdrip/Models/Calibration.java]: Any reason for the order change of the adjusting recent readings?

File: app/src/main/java/com/eveningoutpost/dexdrip/Home.java (hunk 2 above)
Comment (kskandispersonal) [inline app/src/main/java/com/eveningoutpost/dexdrip/Home.java]: Yes, I'm afraid I never tested this.  I think this was existing code when I joined.

File: app/src/main/java/com/eveningoutpost/dexdrip/Tables/CalibrationDataTable.java [hunk 5]
```diff
@@ -118,7 +118,7 @@ public boolean onLongClick(View v) {
                         public void onClick(DialogInterface dialog, int which) {
                             switch (which){
                                 case DialogInterface.BUTTON_POSITIVE:
-                                    calibration.invalidate();
```
Comment (jamorham) [inline app/src/main/java/com/eveningoutpost/dexdrip/Tables/CalibrationDataTable.java]: this maybe has to be approached with caution, but tracing it through it should be an improvement. Possibly as it is from interactive it should have the from_interactive set to true but we can leave that for now.

File: app/src/main/java/com/eveningoutpost/dexdrip/Home.java (hunk 2 above)
Comment (AdrianLxM) [inline app/src/main/java/com/eveningoutpost/dexdrip/Home.java]: It never had mattered as there where only the two options in the menu and this was the second one.
Good example why to write it anyway - to not fall into the trap later :) 
Nice catch

File: app/src/main/java/com/eveningoutpost/dexdrip/Models/BloodTest.java (hunk 3 above)
Comment (kskandispersonal) [inline app/src/main/java/com/eveningoutpost/dexdrip/Models/BloodTest.java]: Probably you made the fix in master, and I didn't pull it down yet, but made it my local repo.

File: app/src/main/java/com/eveningoutpost/dexdrip/Models/Calibration.java (hunk 4 above)
Comment (kskandispersonal) [inline app/src/main/java/com/eveningoutpost/dexdrip/Models/Calibration.java]: Yes, the new calibration should be created on the watch before the BgReading gets associated to it.

File: app/src/main/java/com/eveningoutpost/dexdrip/Home.java (hunk 2 above)
Comment (kskandispersonal) [inline app/src/main/java/com/eveningoutpost/dexdrip/Home.java]: Oh, @AdrianLxM, you are right, I should have noticed it when I added additional cases:(

Comment (AdrianLxM) [inline app/src/main/java/com/eveningoutpost/dexdrip/Home.java]: Sorry if that came out wrong. That was not what I've meant.

**Comment by jamorham** [comment]
Okay, is this still ready for merge with the latest changes?

**Comment by kskandispersonal** [comment]
Yes.  I made a few last minute changes realizing Treatments.systimestamp wasn't being set which effected population of the treatments new data table view.

**Pull request merged**